### PR TITLE
DHFPROD-7645:Include timezone for parseDate() response

### PR DESCRIPTION
**Pull request by srinathgit**
### Description
Include timezone for parseDate() response
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x]  PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit

- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/mapping/core-functions-impl.xqy [hunk 1]
```diff
@@ -174,10 +174,12 @@ declare private function parse-non-standard-date($value as xs:string?, $pattern
       let $month := map:get($MONTH-MAP, fn:upper-case(fn:substring($value, 1, 3)))
       let $day := fn:substring($value, 5, 2)
       let $year := fn:substring($value, 8, 4)
-      return fn:string-join(($year, $month, $day), "-")
+      let $log := xdmp:log(fn:string-join(($year, $month, $day), "-"))
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/mapping/core-functions-impl.xqy:177]: Should remove this and the log stmt below too

File: marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/builtins/steps/mapping/entity-services/runMapping.sjs [hunk 2]
```diff
@@ -13,7 +13,7 @@ function runMapping () {
   let instance = result.envelope.instance;
   try {
     assertions.push(
-      test.assertEqual('2019-12-07', fn.string(instance.OrderType.purchaseDate)),
+      test.assertEqual(xs.string(xs.date(`2019-12-07${serverTimezone}`)), fn.string(instance.OrderType.purchaseDate)),
```
Comment (rjrudin) [inline marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/builtins/steps/mapping/entity-services/runMapping.sjs:16]: No need to change, but it's good to make a function for something like this which is duplicated a couple times - e.g. 

    test.assertEqual(makeDate("2019-12-07"), ...)

Where "makeDate" would handle the timezone bit and calling xs.string/xs.date.

**Review by rjrudin — CHANGES_REQUESTED** [review]
Looks good, just need to remove log statements

File: marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/builtins/steps/mapping/entity-services/runMapping.sjs (hunk 2 above)
Comment (srinathgit) [reply marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/builtins/steps/mapping/entity-services/runMapping.sjs]: It's the same value so I used a variable.

**Review by rjrudin — APPROVED** [review]

**Review by ryanjdew — APPROVED** [review]

**Review by rjrudin — APPROVED** [review]

**Pull request merged**